_pytest/test_multitenancy.py

```diff
@@ -28,98 +34,123 @@ def app(component_builder):
     )
 ])
-def test_get_parse_invalid_model(client, response_test):
-    response = client.get(response_test.endpoint)
+def test_get_parse_invalid_model(stub, response_test):
+    response = stub(requests.get)(response_test.endpoint)
+    rjs = response.json()
     assert response.status_code == 404
-    assert response.json.get("error").startswith(response_test.expected_response["error"])
+    assert rjs.get("error").startswith(response_test.expected_response["error"])
 
 
 @pytest.mark.parametrize("response_test", [
     ResponseTest(
         "/parse",
         {"entities": [], "intent": "affirm", "text": "food"},
-        payload={"q": "food", "model": "one"}
+        payload={"q": "food", "model": "test_model_mitie"}
     ),
     ResponseTest(
         "/parse",
         {"entities": [], "intent": "restaurant_search", "text": "food"},
-        payload={"q": "food", "model": "two"}
+        payload={"q": "food", "model": "test_model_mitie_sklearn"}
     ),
     ResponseTest(
         "/parse",
         {"entities": [], "intent": "restaurant_search", "text": "food"},
-        payload={"q": "food", "model": "three"}
+        payload={"q": "food", "model": "test_model_spacy_sklearn"}
     ),
 ])
-def test_post_parse(client, response_test):
-    response = client.post(response_test.endpoint,
-                           data=json.dumps(response_test.payload), content_type='application/json')
+def test_post_parse(stub, response_test):
+    response = stub(requests.post)(response_test.endpoint, json=response_test.payload)
+    rjs = response.json()
     assert response.status_code == 200
-    assert all(prop in response.json for prop in ['entities', 'intent', 'text'])
+    assert all(prop in rjs for prop in ['entities', 'intent', 'text'])
 
 
 @pytest.mark.parametrize("response_test", [
     ResponseTest(
         "/parse",
-        {"error": "No model found with alias 'default'. Error: Failed to load model metadata. "},
+        {"error": "No agent found with name 'default_agent'."},
         payload={"q": "food"}
     ),
     ResponseTest(
         "/parse",
-        {"error": "No model found with alias 'umpalumpa'. Error: Failed to load model metadata. "},
+        {"error": "No agent found with name 'umpalumpa'."},
         payload={"q": "food", "model": "umpalumpa"}
     ),
 ])
-def test_post_parse_invalid_model(client, response_test):
-    response = client.post(response_test.endpoint,
-                           data=json.dumps(response_test.payload), content_type='application/json')
+def test_post_parse_invalid_model(stub, response_test):
+    response = stub(requests.post)(response_test.endpoint, json=response_test.payload)
+    rjs = response.json()
     assert response.status_code == 404
-    assert response.json.get("error").startswith(response_test.expected_response["error"])
+    assert rjs.get("error").startswith(response_test.expected_response["error"])
 
 
-if __name__ == '__main__':
+def train_models():
```

Comment:
Is this still needed? Even before this part was never called. Moreover calling this part in the `stub` fixture makes the whole test stall (during a call to a MITIE function), whereas individually it works fine.